Lib/test/test_format.py

```diff
@@ -519,5 +519,27 @@ def test_with_an_underscore_and_a_comma_in_format_specifier(self):
         with self.assertRaisesRegex(ValueError, error_msg):
             '{:_,}'.format(1)
 
+    @support.cpython_only
+    def test_better_error_message(self):
+        with self.assertRaises(ValueError) as e:
+            "{l:%M}".format(l=complex(12))
+        self.assertEqual(str(e.exception),
+            "Invalid format specifier: '{l:%M}' for object of type 'complex'")
```

Comment:
I would expect this to print:

             "Invalid format specifier: '%M' for object of type 'complex'"

I'll have to research what's going on.